### PR TITLE
Fix branch switcher conflict with `default-branch-button`

### LINKED ISSUE #4240: `default-branch-button` feature breaks the branch switcher
Here's a GIF:

And a link to the place where this can be reproduced:
https://github.com/File-New-Project/EarTrumpet/blob/2.1.8.0/README.md

### PR DESCRIPTION
**Pull request by cheap-glitch**
Fixes #4240 

## Test URLs

Go to https://github.com/sindresorhus/refined-github/tree/hotfix, click on the branch switcher and select an option above the readme header.

## Screenshot

https://user-images.githubusercontent.com/46634000/119224239-c89a9c00-bafd-11eb-9f2b-3a97987426e5.mp4

<p>&nbsp;</p>

The culprit:
image

### CODE REVIEW

**Comment by fregante** [comment]
Ohhhh wow. That makes sense. But now I wonder why we’re not seeing this anywhere else on native buttons. Maybe they avoid it with a custom class?

**Comment by fregante** [comment]
> why we’re not seeing this anywhere else on native buttons

I looked around the website a bit and `.BtnGroup` elements are rather rare, especially the ones with a dropdown inside, and this is probably the only situation where the z-index change causes this issue anyway.

**Comment by fregante** [comment]
Can you try adding an inline CSS to the group?

`groupButtons` returns the element, so it's probably:

```ts
const groupedButtons = groupButtons();
groupedButtons.style.zIndex = '…' // comment

**Comment by cheap-glitch** [comment]
> Can you try adding an inline CSS to the group?

It doesn't work, the z-index of the wrapper doesn't affect the buttons since it's overridden by the focus selector. But we can do:
```js
branchSelector.style.zIndex = 'auto';
branchSelector.parentElement!.style.zIndex = 'auto';

```

File: source/features/default-branch-button.tsx [hunk 1]
```diff
@@ -51,6 +51,10 @@ async function init(): Promise<false | void> {
 	branchSelector.parentElement!.before(defaultLink);
 	groupButtons([defaultLink, branchSelector.parentElement!]).classList.add('d-flex');
 	branchSelector.style.float = 'none'; // Pre "Repository refresh" layout
+
+	// Fix #4240
+	branchSelector.style.zIndex = 'auto';
+	branchSelector.parentElement!.style.zIndex = 'auto';
```
Comment (fregante) [inline source/features/default-branch-button.tsx:57]: The first line doesn't seem necessary. Also this order makes aligns the two `branchSelector.parentElement!.` calls

```suggestion
	branchSelector.parentElement!.style.zIndex = 'auto'; // For #4240
	groupButtons([defaultLink, branchSelector.parentElement!]).classList.add('d-flex');
	branchSelector.style.float = 'none'; // Pre "Repository refresh" layout
```

**Comment by fregante** [comment]
Thank you!

**Pull request merged**